BillTrack.Application/Services/SqsPublisher.cs

```diff
@@ -44,6 +48,34 @@ public async Task PublishMessageAsync<T>(string queueName, T message) where T :
             }
         };
 
-        await _sqs.SendMessageAsync(request);
+        try
+        {
+            await _sqs.SendMessageAsync(request);
+        }
+        catch (Exception ex)
+        {
+            throw;
+        }
+    }
+
+    private async Task<string> GetQueueUrlAsync(string queueName)
+    {
+        if (_cachedQueueUrl.TryGetValue(queueName, out var cachedUrl))
+        {
+            return cachedUrl;
+        }
+
+        try
+        {
+            var response = await _sqs.GetQueueUrlAsync(queueName);
+            var queueUrl = response.QueueUrl;
+            _cachedQueueUrl[queueName] = queueUrl;
+            return queueUrl;
```

Comment:
A `try` block should be as short as possible. When an exception is thrown, .NET will build the entire detailed call stack for all statements in the block.

```suggestion
GetQueueUrlAsyncResposeType response = null;

try {
    response = await _sqs.GetQueueUrlAsync(queueName);
}
catch (QueueDoesNotExistException)
{
    _cachedQueueUrl.TryRemove(queueName, out _);
    throw;
}

if (null == response || string.IsNullOrEmpty(response.QueueUrl) {
    throw new QueueDoesNotExistException("Error during a queue URL resolution");
}

var queueUrl = response.QueueUrl;
_cachedQueueUrl[queueName] = queueUrl;
return queueUrl;
```